addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -629,8 +628,6 @@ export class HtmlField extends Component {
         const rating = shouldToggleOff ? 0 : previousStars.length + 1;
 
         let starsId = $(node).parent().attr('id');
-        starsId = starsId && starsId.replace('checkId-', '');
-        starsId = parseInt(starsId || '0');
```

Comment:
@Zinston Could you have a look at those changes please ? I'm very confused by the removal of those lines as it seems to partially undo what was done at https://github.com/odoo/odoo/pull/83137/commits/1f61f0d397eee6be2cbcf4042a1c15fa07063fb2#diff-c1e8fb6ac2e4699027311d5c71fb2679f25ca75fbb8cbac00f79d77b55d7a13dR526.

Reply:
This is confusing the hell out of me.

With these changes:

Say we have `<star id="checkId-1"/>`.
1. `let starsId = $(node).parent().attr('id');` so `starsId === 'checkId-1'` (and `let` should be `const`).
2. We do an RPC to `/web_editor/stars` with `starsId = 'checkId-1'`
3. In `update_stars`: `stars_widget = htmlelem.find(".//span[@id='checkId-%s']" % starsId)`
4. That's equivalent to `stars_widget = htmlelem.find(".//span[@id='checkId-checkId-1']")`
5. Therefore `stars_widget is None`
6. We return just below in `if stars_widget is None: return value`.

So we never check the star. I don't see how this could ever work.